### PR TITLE
Add workflow for wheels

### PR DESCRIPTION
**Pull request by sevagh**
This PR replaces PR #4 (so we can preserve the original while we work on both side by side).

### CODE REVIEW

File: .github/workflows/wheels-pure.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    steps:
+    - name: checkout code repo
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0 # unshallow fetch for setuptools-scm
+        persist-credentials: false
+
+    - name: Build pure python wheel with pip wheel
+      run: python-cp38 -m pip wheel -w ./dist ${{ inputs.package-dir }} --no-deps
```
Comment (sevagh) [inline .github/workflows/wheels-pure.yml:58]: Note: this is my own symlink, maybe think about doing it differently

File: .github/workflows/wheels-pure.yml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        RAPIDS_PY_VERSION: "3.8"
+        AWS_ACCESS_KEY_ID: ${{ secrets.RAPIDSAI_GHA_AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.RAPIDSAI_GHA_AWS_SECRET_ACCESS_KEY }}
+
+  wheel-test:
+    name: wheel test pure
+    runs-on: [self-hosted, linux, amd64, gpu-v100-495-1]
+    needs: wheel-build
+    container:
+      image: "rapidsai/citestwheel_x86_64"
```
Comment (sevagh) [inline .github/workflows/wheels-pure.yml:77]: Merge cibuildwheel/citestwheel containers (they're mostly the same)

Comment (sevagh) [reply .github/workflows/wheels-pure.yml:77]: It takes ~1 minute to download the runtime cibw launcher, and ~3-4 minutes to download the devel citestwheel tester - keeping them separate is fine.

**Review by vyasr — APPROVED** [review]
I'm pretty happy with this workflow at this stage. There are a couple of variables that are probably extraneous right now but we will need later like docker flags, and I wonder if maybe the cuda-suffix shouldn't even be configurable, but I think we're nitpicking now so let's merge and just iterate as we go. 

Thanks for working through all the issues here!

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**